### ISSUE TITLE
Single GPU training

### ISSUE THREAD

**Issue by petercmh01** [wpeebles/gangealing#18]
Did anyone manage to run training on a single GPU? I keep getting **torch.distributed.elastic.multiprocessing.errors.child failed error** when using one GPU to train with this following training script: 

torchrun --nproc_per_node=1   /path/train.py \
--ckpt cat --load_G_only --padding_mode border --vis_every 500 --ckpt_every 10000 \
--iter 100000 --tv_weight 2500 --loss_fn lpips --exp-name lsun_cats_lpips --real_data_path /my/train_data/

Thanks in advance! 

Edit: I tried to turn off the distributed mode but a math value error pops out in here https://github.com/wpeebles/gangealing/blob/739da2a25de62702d54d83fad6b644646512039c/utils/annealing.py#L42

says Value error math domain error

**Comment by wpeebles** [comment]
Hmm, that's strange. I don't think I've encountered any issues training with one GPU before. Can you try replacing the `torchrun --nproc_ner_node=1` with `python`? Also, what version of Python are using?

**Comment by petercmh01** [comment]
I'm running it on colab using: 

python 3.7.13
torch 1.11.0
cuda 11.2

**Comment by petercmh01** [comment]
Hi thre! 
I was able to run the training by using this script

python /content/gangealing/train.py \
--ckpt .../myckpt/020000.pt --load_G_only --vis_every 500 --ckpt_every 10000 \
--iter 100000 --tv_weight 2500 --loss_fn lpips --exp-name lsun_cats_lpips --real_data_path  .../train_folder/

and I had to manually turn off all the distributed setting (i.e., set distributed = False) and I had to set the num_cycles in utils/annealing.pyto an integer. 

I'm using colab with python 3.7, pytorch 1.11.0, cuda 11.2

Does the number of cycle impact the training a lot? 

-------------
Edit: turns out the learning rate cycle error was my iteration number is not big enough. By using iteration = 1500000 it run with no error.

**Comment by wpeebles** [comment]
Great! To answer your follow-up question, I haven't carefully ablated the impact of the number of learning rate cycles, but my guess is that it shouldn't matter too much. 

By the way, when training on 1 gpu you will likely get better results when using gradient accumulation to get a larger effective batch size (see issue https://github.com/wpeebles/gangealing/issues/10). The `--batch` argument controls the per-GPU batch size. Ideally, total batch size should be at least 40 or so, but anything greater than 5 should help improve results.
